### PR TITLE
ALTAPPS-567: Shared fix streak freeze state synchronization

### PR DESCRIPTION
**Pull request by vladkash**
**YouTrack Issues**:
[#ALTAPPS-567](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-567)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
Added synchronization for streak freeze state when hypercoins balance is changed. More details in task [comment](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-567/Shared-1-day-streak-freeze-button-is-visible-at-a-time-while-I-already-have-one-strike-freeze#focus=Comments-74-219184.0-0)

### CODE REVIEW

**Review by XanderZhu — APPROVED** [review]

File: shared/src/commonMain/kotlin/org/hyperskill/app/profile/presentation/ProfileFeature.kt [hunk 1]
```diff
@@ -51,7 +51,10 @@ interface ProfileFeature {
         ) : StreakFreezeState
 
         @Serializable
-        data class NotEnoughGems(val price: Int) : StreakFreezeState
+        data class NotEnoughGems(
+            val streakFreezeProductId: Long,
```
Comment (ivan-magda) [inline shared/src/commonMain/kotlin/org/hyperskill/app/profile/presentation/ProfileFeature.kt:55]: Where `streakFreezeProductId` is used?

Comment (ivan-magda) [reply shared/src/commonMain/kotlin/org/hyperskill/app/profile/presentation/ProfileFeature.kt:55]: OK, found `ProfileFeature.StreakFreezeState.CanBuy(state.streakFreezeState.streakFreezeProductId, state.streakFreezeState.price)`

File: shared/src/commonMain/kotlin/org/hyperskill/app/profile/presentation/ProfileReducer.kt [hunk 2]
```diff
@@ -62,7 +62,20 @@ class ProfileReducer : StateReducer<State, Message, Action> {
...
+                        streakFreezeState = if (
+                            state.streakFreezeState is ProfileFeature.StreakFreezeState.NotEnoughGems &&
+                            state.streakFreezeState.price <= message.hypercoinsBalance
+                        )  {
+                            ProfileFeature.StreakFreezeState.CanBuy(state.streakFreezeState.streakFreezeProductId, state.streakFreezeState.price)
+                        } else if (
+                            state.streakFreezeState is ProfileFeature.StreakFreezeState.CanBuy &&
+                            state.streakFreezeState.price > message.hypercoinsBalance
+                        ) {
+                            ProfileFeature.StreakFreezeState.NotEnoughGems(state.streakFreezeState.streakFreezeProductId, state.streakFreezeState.price)
```
Comment (ivan-magda) [inline shared/src/commonMain/kotlin/org/hyperskill/app/profile/presentation/ProfileReducer.kt:75]: Exceeds `max_line_length = 120`.

Reformat code to this:
```kotlin
streakFreezeState = if (
    state.streakFreezeState is ProfileFeature.StreakFreezeState.NotEnoughGems &&
    state.streakFreezeState.price <= message.hypercoinsBalance
) {
    ProfileFeature.StreakFreezeState.CanBuy(
        state.streakFreezeState.streakFreezeProductId,
        state.streakFreezeState.price
    )
} else if (
    state.streakFreezeState is ProfileFeature.StreakFreezeState.CanBuy &&
    state.streakFreezeState.price > message.hypercoinsBalance
) {
    ProfileFeature.StreakFreezeState.NotEnoughGems(
        state.streakFreezeState.streakFreezeProductId,
        state.streakFreezeState.price
    )
} else {
    state.streakFreezeState
}
```

Comment (ivan-magda) [reply shared/src/commonMain/kotlin/org/hyperskill/app/profile/presentation/ProfileReducer.kt:75]: You can enable "Visual guides" with the value of "120" in your IDE

Comment (XanderZhu) [reply shared/src/commonMain/kotlin/org/hyperskill/app/profile/presentation/ProfileReducer.kt:75]: @ivan-magda 
We can enable max_line_length setting in ktlint rules.
A the moment it is manually disabled.

Comment (ivan-magda) [reply shared/src/commonMain/kotlin/org/hyperskill/app/profile/presentation/ProfileReducer.kt:75]: Yes, we have task about this ALTAPPS-490

**Review by ivan-magda — APPROVED** [review]

**Pull request merged**